src/Psalm/Internal/Type/TypeParser.php

```diff
@@ -830,19 +830,24 @@ private static function getTypeFromGenericTree(
                 throw new TypeParseTreeException('Union types are not allowed in int range type');
             }
 
-            if ($param0_union_types[0] instanceof TNamedObject &&
-                $param0_union_types[0]->value === TIntRange::BOUND_MAX
-            ) {
-                throw new TypeParseTreeException("min bound for int range param can't be 'max'");
+            if (!($param0_union_types[0] instanceof TNamedObject) && !($param0_union_types[0] instanceof TLiteralInt) ||
+                !($param1_union_types[0] instanceof TNamedObject) && !($param1_union_types[0] instanceof TLiteralInt)) {
+                throw new TypeParseTreeException('Unsupported parameter');
             }
-            if ($param1_union_types[0] instanceof TNamedObject &&
-                $param1_union_types[0]->value === TIntRange::BOUND_MIN
-            ) {
-                throw new TypeParseTreeException("max bound for int range param can't be 'min'");
+
+            if ($param0_union_types[0] instanceof TNamedObject
+                && $param0_union_types[0]->value !== TIntRange::BOUND_MIN) {
+                throw new TypeParseTreeException('Unknown named object as a min boundary');
+            }
+
+            if ($param1_union_types[0] instanceof TNamedObject
+                && $param1_union_types[0]->value !== TIntRange::BOUND_MAX) {
+                throw new TypeParseTreeException('Unknown named object as a max boundary');
```

Comment:
I would change these messages, [here](https://psalm.dev/r/30ef078e0f) `Foo` isn't an unknown named object, it's known it's just incorrect.

Reply:
Fixed!